modules/v1/workspaces/workspaceBehaviours/TimelineUrlBehaviour.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+
+namespace app\modules\v1\workspaces\workspaceBehaviours;
+
+use app\modules\v1\workspaces\models\TimelinePost;
+use yii\base\Behavior;
+use yii\db\ActiveRecord;
+
+class TimelineUrlBehaviour extends Behavior
+{
+    public $timelinePost;
+
+    public function events()
+    {
+        return [
+            ActiveRecord::EVENT_AFTER_INSERT => 'afterInsert',
+            ActiveRecord::EVENT_AFTER_UPDATE => 'afterInsert',
+        ];
+    }
+
+    public function afterInsert() {
+        if(strlen($this->timelinePost->description !== 0)) {
+            $description = $this->timelinePost->description;
+            $description = strip_tags(preg_replace("((<\/)\w+(>))", "\n", $description));
+            $descriptions = preg_split("/[\n\s,]+/", $description);
+
+            $regex = "/^(http|https):\/\/[^\s]+/";
+            foreach ($descriptions as $i) {
```

Comment:
You are iterating char by char in the description which obviously is not necessary.
You just need to write preg_replace once.

Reply:
With case of multiline description, tags are adding up `<p>description</p><p>newline</p>.` After splitting, it becomes an array of "clean descriptions" and I iterate over them and apply link regex